packages/amplify-cli/src/commands/diagnose.ts

```diff
@@ -87,9 +87,16 @@ const showLearnMore = (showOptOut: boolean): void => {
 const zipSend = async (context: Context, skipPrompts: boolean, error: Error | undefined): Promise<void> => {
   const choices = ['Generate report', 'Nothing'];
   if (!skipPrompts) {
-    const diagnoseAction = await prompter.pick('What would you like to do?', choices);
-    if (diagnoseAction !== choices[0]) {
-      return;
+    if (DebugConfig.Instance.promptSendReport()) {
+      const result = await prompter.yesOrNo('Help improve Amplify CLI by sharing non sensitive configurations on failures', false);
```

Comment:
Looks like we've already got a prompt on line 37, why do we need to add this line to `zipSend` again? https://github.com/aws-amplify/amplify-cli/pull/12710/files#diff-71cabcaa3379af3b9458f6802b25c65f6c00768606d5ef7f8c010094d8720aa1R37